src/clib/pio_spmd.c

```diff
@@ -291,7 +302,7 @@ int pio_swapm(void *sendbuf, int *sendcounts, int *sdispls, MPI_Datatype *sendty
              * used to choose between mpi_irsends and mpi_isends - the default
              * is still mpi_irsend
              */
-            if (isend)
+            if (handshake && isend)
```

Comment:
This change doesn't make any sense to me.  The code for isend here is the same whether or not handshake is true, except that you removed the ifdef USE_MPI_ISEND_FOR_FC